File: go/2808.go
package main

func minimumSeconds(nums []int) int {
	pos_lists := map[int][]int{}
	for index, num := range nums {
		pos_lists[num] = append(pos_lists[num], index)
	}
	n := len(nums)
	res := len(nums)
	for _, pos_list := range pos_lists {
		pos_list = append(pos_list, pos_list[0]+n)
		max_gap := pos_list[1] - pos_list[0]
		for i := 0; i < len(pos_list)-1; i++ {
			max_gap = max(max_gap, pos_list[i+1]-pos_list[i])
		}
		res = min(res, max_gap/2)
	}
	return res
}

File: python/2808.py
from collections import defaultdict
from itertools import pairwise
from typing import List

class Solution:
    def minimumSeconds(self, nums: List[int]) -> int:
        pos = defaultdict(list)
        for index, num in enumerate(nums):
            pos[num].append(index)
        ans = n = len(nums)
        for pos_list in pos.values():
            pos_list.append(pos_list[0] + n)
            pos_max_step = max(j - i for i, j in pairwise(pos_list)) // 2
            ans = min(ans, pos_max_step)
        return ans